### PR TITLE
Support add/remove label to/from artifact

### PR DESCRIPTION
**Pull request by ywk253100**
This commit add supporting for adding/removing label to/from artifacts and populates labels when listing artifacts

Signed-off-by: Wenkai Yin <yinw@vmware.com>

### CODE REVIEW

File: src/api/artifact/controller.go [hunk 1]
```diff
@@ -407,30 +429,46 @@ func (c *controller) assembleTag(ctx context.Context, tag *tm.Tag, option *TagOp
...
 	if err != nil {
 		log.Error(err)
-		return false
+		return
 	}
-	return matched
+	tag.Immutable = matched
+}
+
+func (c *controller) populateScanOverview(ctx context.Context, art *Artifact) {
```
Comment (steven-zou) [inline src/api/artifact/controller.go]: Is it possible to define a standard populate interface with kinds of implementations for such cases?

```
func (c *controller) populateScanOverview(ctx context.Context, art *Artifact) {
	// TODO implement
}

func (c *controller) populateSignature(ctx context.Context, art *Artifact) {
	// TODO implement
	// TODO populate signature on artifact or tag level?
}	}
```

File: src/api/artifact/model.go [hunk 2]
```diff
@@ -82,14 +84,27 @@ func (a *Artifact) ToSwagger() *models.Artifact {
 			Href:     link.HREF,
 		}
 	}
+	for _, label := range a.Labels {
+		art.Labels = append(art.Labels, &models.Label{
```
Comment (steven-zou) [inline src/api/artifact/model.go]: Should the labels in the artifact be a full one (with all properties)?

File: src/pkg/label/manager.go [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+	ierror "github.com/goharbor/harbor/src/internal/error"
+	"github.com/goharbor/harbor/src/pkg/q"
+	"time"
+)
+
+// Mgr is a global instance of label manager
+var Mgr = New()
+
+// Manager manages the labels and references between label and resource
+type Manager interface {
```
Comment (steven-zou) [inline src/pkg/label/manager.go]: label's CRUD and reference management should be two parts, right? 
label mgr for label CURD
labelMgr.ReferenceMgr() returns the ref mgr that is responsible for reference management?

File: src/api/artifact/model.go (hunk 2 above)
Comment (ywk253100) [reply src/api/artifact/model.go]: Yes, almost all of the properties is needed when rendering the UI

File: src/api/artifact/controller.go (hunk 1 above)
Comment (ywk253100) [reply src/api/artifact/controller.go]: Issue https://github.com/goharbor/harbor/issues/10741 opened to track the work

File: src/pkg/label/manager.go (hunk 3 above)
Comment (ywk253100) [reply src/pkg/label/manager.go]: I'll submit another PR to cover this part for both label manager and artifact manager

File: src/api/artifact/model.go (hunk 2 above)
Comment (steven-zou) [reply src/api/artifact/model.go]: that's weird

**Pull request merged**